Document new modbus configuration parameter

reiffert2 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Document the new modbus specific configuration parameter restart_delay


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/130038
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new optional configuration parameter `restart_delay` for the Modbus integration, allowing users to specify a delay before re-establishing a connection.
	- Updated documentation to include examples of the new parameter, enhancing user awareness and understanding of its usage.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

reiffert2 committed:
Update modbus.markdown

Document restart_delay configuration option

reiffert2 committed:
Update modbus.markdown

c0ffeeca7 commented on source/_integrations/modbus.markdown:
```diff
@@ -56,6 +56,13 @@ name:
+restart_delay:
+  description: "Time to delay re-establishing a connection after connection close.
+  Some modbus devices need a delay of typically 1-2 seconds after an existing connection is closed before creating a new connection.
+  If a device does not respond to messages after reloading yaml files, then try this parameter."
```
```suggestion
  If a device does not respond to messages after reloading YAML files, then try this parameter."
```

c0ffeeca7 committed:
fix typo

reiffert2 committed:
Merge branch 'next' into next

frenck reviewed (dismissed):
✅ Approved. Can be merged as soon as the parent PR gets merged.

reiffert2 committed:
Merge branch 'next' into next

reiffert2 committed:
Update modbus.markdown

Updating markdown after moving the default from 0 seconds to 1 second after suggestion from @frenck and @rytilahti

reiffert2 committed:
Merge branch 'next' into next
